workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/WorkflowRendering.kt

```diff
@@ -101,34 +95,9 @@ public fun WorkflowRendering(
  * bit simpler since we don't need to worry about attachment state.
  */
 @Composable private fun rememberChildLifecycleOwner(): LifecycleOwner {
-  val lifecycleOwner = remember {
-    object : LifecycleOwner {
-      val registry = LifecycleRegistry(this)
-      override val lifecycle: Lifecycle
-        get() = registry
-    }
-  }
   val parentLifecycle = LocalLifecycleOwner.current.lifecycle
-
-  DisposableEffect(parentLifecycle) {
-    val parentObserver = LifecycleEventObserver { _, event ->
-      // Any time the parent lifecycle changes state, perform the same change on our lifecycle.
-      lifecycleOwner.registry.handleLifecycleEvent(event)
-    }
-
-    parentLifecycle.addObserver(parentObserver)
-    onDispose {
-      parentLifecycle.removeObserver(parentObserver)
-
-      // If we're leaving the composition it means the WorkflowRendering is either going away itself
-      // or about to switch to an incompatible rendering – either way, this lifecycle is dead. Note
-      // that we can't transition from INITIALIZED to DESTROYED – the LifecycleRegistry will throw.
-      // WorkflowLifecycleOwner has this same check.
-      if (lifecycleOwner.registry.currentState != INITIALIZED) {
-        lifecycleOwner.registry.currentState = DESTROYED
-      }
-    }
+  val lifecycleOwner = remember(parentLifecycle) {
+    ComposeLifecycleOwner(parentLifecycle)
```

Comment:
(See comment on other file)
```suggestion
    ComposeLifecycleOwner(parentLifecycle).also(parentLifecycle::addObserver)
```